msamp/nn/linear.py

```diff
@@ -16,14 +16,15 @@ class FP8Linear(ScalingModule):
     DEFAULT_WINDOW_SIZE = 16
     DEFAULT_WGRAD_WINDOW_SIZE = 1
 
-    def __init__(self, in_features, out_features, use_bias=True, weight_qtype=Dtypes.kfloat16):
+    def __init__(self, in_features, out_features, use_bias=True, weight_qtype=Dtypes.kfloat16, bias_type=torch.float32):
```

Comment:
It is. Without this change, the example can't run successfully. Here is the detail description of this bug:
If fp16 is enabled in deep speed, all the parameters of the model will be cast to fp16. But after linear modules are replaced by FP8Linear, bias will be changed from fp16 to fp32. In FP8Optimizer, all the parameters of non-ScalingTensor will be cast to FP32 because of flattern/unflattern. But Conv2d with fp32 parameters can't accept fp16 input and will get an error "Input type (c10::Half) and bias type (float) should be the same".  So the way of fixing this bug is straightforward, just don't change the dtype of bias when replacing Linear with FP8Linear.